src/libstd/sys_common/mutex.rs

```diff
@@ -17,6 +17,7 @@ impl Mutex {
     /// Also, until `init` is called, behavior is undefined if this
     /// mutex is ever used reentrantly, i.e., `raw_lock` or `try_lock`
     /// are called by the thread currently holding the lock.
+    #[rustc_const_stable(feature = "const_sys_mutex_new", since = "1.0.0")]
```

Comment:
Out of curiosity where did this come from?

Reply:
It is a bit convoluted (and I don't 100% understand the const stability infrastructure).

`Mutex::new` is used inside a const fn [here](https://github.com/rust-lang/rust/blob/0c03aee8b81185d65b5821518661c30ecdb42de5/src/libstd/io/lazy.rs#L22) for `std::io::lazy::Lazy::new`.  The const-stability checks are tied to the overall stability checks.  When libstd is marked unstable (via restricted-std), somehow this propagates to const functions as well.  Surprisingly this is the only const fn that tripped this issue. The specific error is:

> error[E0723]: can only call other `const fn` within a `const fn`, but `const sys_common::mutex::Mutex::new` is not stable as `const fn`

I believe when a `const fn` has no stability attributes, it inherits whatever the crate (or module?) stability is.  (Sorry, it's been months since I worked on this, so my memory is a bit fuzzy.)

Reply:
Ah ok no worries, I was a little curious but I'd believe that it falls out of *something*, and it's internal anyway so it doesn't matter too much.